docs/troubleshooting.md

```diff
@@ -0,0 +1,202 @@
+
+This guide explains how to troubleshoot common issues related to OSDFIR Infrastructure's Helm chart.
+
+### Common issues
+
+The following are the most common issues that users face when dealing with deployments:
+
+* Pods failing to start up
+* Unable to access services
+* Persistence Volumes (PVs) failing to provision
+* An issue with the underlying Helm chart
+
+### Troubleshoot Pods
+
+To check if the status of your pods is healthy, the easiest way is to run the *kubectl get pods* command. After that, you can use *kubectl describe* and *kubectl logs* to obtain more detailed information.
+
+Once you run *kubectl get pods*, you may find your pods showing any of the following statuses:
+
+* *Pending* or *CrashLoopBackOff*
+* *ImagePullBackOff* or *ErrImagePull*
+* *Init:CrashLoopBackOff* or *Init:Error*
+
+If you see any of the statuses above, this means that the pod could not be scheduled on a node. Usually, this is because
+of insufficient CPU or memory resources. It may also come up due to a network related issue or provisioning a volume.
+
+> IMPORTANT: Please wait a few minutes for errors to resolve when installing a Helm chart for the first time as it is likely still provisioning the necessary resources.
+
+To confirm the cause, note the pod you want to investigate from *kubectl get pods*, replacing `POD-NAME` with the name of the pod, then run:
+
+```shell
+$ kubectl describe pod POD-NAME
+```
+
+You should see an output message providing some information about why the pod is failing.  Additionally, you can review logs of the pod by running:
+
+```shell
+$ kubectl logs POD-NAME
+```
+
+Another option can be to review logs stored within the pod. To gain shell access to a pod, run:
+
+```shell
+$ kubectl exec -it POD-NAME -- /bin/bash
+```
+
+> TIP: Turbinia logs can be found in `/mnt/turbiniavolume/logs` and Timesketch logs can be found in `/var/log/timesketch`.
+
+When your pod status is *ImagePullBackOff* or *ErrImagePull*, this means that the pod could not run because it could not pull the image. This typically occurs when there is either an invalid Image or the Kubernetes cluster does not have Internet connectivity. Try pulling the image manually on the node using *docker pull*. For example, to manually pull an image from Docker Hub, use the command below by replacing IMAGE with the image ID:
+
+```shell
+$ docker pull IMAGE
+```
+
+> TIP: When using a private GKE cluster in GCP, please ensure you have a [Cloud NAT](https://cloud.google.com/nat/docs/gke-example#gcloud_3) configured with the network router of your cluster so that third party dependencies can be retrieved.
+
+When your pod status is *Init:CrashLoopBackOff* or *Init:Error*, this means that the pod's init container could not run. To investigate, review the logs of the init container, replacing `POD-NAME` with the pod name and `INIT-POD` with the name of the init pod, then run:
```

Comment:
Probable makes more sense to call it INIT-CONTAINER (as it is not a pod)

Reply:
Agreed, updated